### PR TITLE
fix: brew require arch

### LINKED ISSUE #2159: brew: Add explicit OS/arch dependencies to communicate compatibility gracefully
**Describe the bug**

Users attempting to install a program via Homebrew formula generated by GoReleaser on unsupported OS/arch may receive confusing errors and be unable to `brew tap` the whole repo just because it contains invalid formulas with effectively missing `url` (after none of [these conditions](https://github.com/goreleaser/goreleaser/blob/41356ea84d734928493a7342d3c44e4b163237bc/internal/pipe/brew/template.go#L52-L89) are met).

**To Reproduce**

I don't have a full repro case (yet), but we maintain a generator that generates formulas in pretty much the same way as GoReleaser and people have reported failures on Apple Silicon https://github.com/hashicorp/homebrew-tap/issues/134 so I'm working with the assumption that this must affect GoReleaser too, it just probably isn't a common enough problem people run into (yet).

**Expected behavior**

User is presented with a clear error message that their OS/arch is not supported and (most importantly) they're able to still `brew tap` the tap even if it contains formulas which are not supported on their platform.

**Additional context**

I believe we could just set some default dependencies in formulas automatically as we know what's supported (OS/arch) based on the builds anyway.

For example
```ruby
depends_on arch: [:x86_64] if OS.mac?
```
or alternatively
```ruby
on_macos do
  depends_on arch: [:x86_64]
end
```

--- 

Apologies for the slightly vague description and lack of repro case - I'm just currently swamped with other unrelated work and wanted to create this issue to track the problem until I get to troubleshoot it properly and ideally send a patch.

### PR DESCRIPTION
**Pull request by caarlos0**
closes #2159
refs #2261
see https://github.com/Homebrew/discussions/discussions/1595

### CODE REVIEW

File: internal/pipe/brew/template.go [hunk 1]
```diff
@@ -39,6 +41,36 @@ const formulaTemplate = `# typed: false
 {{ if .CustomRequire -}}
 require_relative "{{ .CustomRequire }}"
 {{ end -}}
+
+class SupportedArchRequirement < Requirement
```
Comment (radeksimko) [inline internal/pipe/brew/template.go:45]: Am I right (per your comments in the linked discussion) that for this to work (and Ruby to be satisfied) this extra class would have to be in a separate file?

If so then we'd be making the assumption that the whole tap/repo is basically owned by GoReleaser, which seems like a flawed assumption?

**Review by radeksimko — COMMENTED** [review]
Given the relative silence in the linked discussion and relatively suboptimal available "easy/quick" solution (based on my understanding) I'm wondering if it's better to instead treat this as "upstream bug" and have it fixed upstream, in Homebrew as initially discussed?

https://github.com/Homebrew/discussions/discussions/1595#discussioncomment-816103

**Comment by caarlos0** [comment]
Yep, probably would be cleaner than this workaround...

**Comment by caarlos0** [comment]
since no news on this, will close.

File: internal/pipe/brew/template.go (hunk 1 above)
Comment (ryancurrah) [reply internal/pipe/brew/template.go:45]: For us our tap is managed only by goreleaser.

Comment (caarlos0) [reply internal/pipe/brew/template.go:45]: in any case, this goes to the tap itself, it does not need to be a separated file...

**Pull request closed without merging**